packages/manager/modules/hub/src/components/payment-status-tile/AutorenewService.class.js

```diff
@@ -0,0 +1,63 @@
+import includes from 'lodash/includes';
+import snakeCase from 'lodash/snakeCase';
+
+import { DEBT_STATUS } from './payment-status-tile.constants';
+
+export default class AutorenewService {
+  constructor(service) {
+    Object.assign(this, service);
+
+    this.isExpired = this.status.toLowerCase() === 'expired';
+    this.isResiliated = this.getResiliated();
+    this.hasAutomaticRenewal = this.renew.automatic || this.hasForcedRenew();
+    this.hasDebt = includes(DEBT_STATUS, snakeCase(this.status).toUpperCase());
+    this.hasManualRenew =
+      this.renew.manualPayment || this.renewalType === 'manual';
+    this.hasPendingResiliation = this.pendingResiliation();
+    this.renewType = this.getRenew();
+    this.isOneShot = this.renewalType === 'oneShot';
+  }
+
+  getRenew() {
+    if (this.hasManualRenew) {
+      return 'manualPayment';
+    }
+
+    if (this.renew.deleteAtExpiration && !this.isResiliated) {
+      return 'delete_at_expiration';
+    }
+
+    if (this.isResiliated) {
+      return 'expired';
+    }
+
+    if (this.hasAutomaticRenewal) {
+      return 'automatic';
+    }
+
+    return 'manualPayment';
+  }
+
+  getResiliated() {
+    return (
+      this.isExpired ||
+      (new Date().getTime() > new Date(this.expiration).getTime() &&
```

Comment:
You should not rely on a date provided by the frontend to know if a service is expired. The api should provide you the information (or you could use /auth/time but i don't think the front have to to this kind of computations)

Reply:
This https://github.com/ovh/manager/pull/2315 will avoid duplicating the logic in each app